rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -261,30 +140,17 @@ Node::create_subscription(
     allocator);
 }
 
-template<typename CallbackType>
-typename rclcpp::timer::WallTimer<CallbackType>::SharedPtr
+template<typename DurationT, typename CallbackT>
+typename rclcpp::timer::WallTimer<CallbackT>::SharedPtr
 Node::create_wall_timer(
```

Comment:
again, just to be consistent, I would also put the `create_wall_timer` into its own file.